docs/proposal-006-build-improvements.md

```diff
@@ -0,0 +1,55 @@
+# Design Proposal #006: Build Improvements
+
+## Current State
+
+At the start of the project we've picked CircleCI as it appears to be a default choice at Weaveworks.
+
+The original solution involved:
+- running `docker build` to create a build image
+- use build image to run tests produce final image
+
+Publishing build and final images hasn't been automated, but some users requested it.
+
+When we switched [to modules](https://github.com/weaveworks/eksctl/pull/917), we have increased the
+compexity of how build image worked. It's hard for a new person to understand it, e.g. for a new
+contributor.
+
+We currently build images in CI and run tests as part of that, we should only run tests as a primary
+task.
+
+Downside of current setup:
+
+- CircleCI requires delegation of access to GitHub
+  - for releases we use a token
+  - we don't have a way to automate workflows that require pushing to the repo
+- CircleCI cache 
+  - it is not very fast, it turns out pulling image with cached dependencies is as fast
+  - it is very specifig to CircleCI, not portable
+  - we do not have a way to controll it
+- the task of building images should not be mixed with the task of running test
+- images are not being automatically pushed to a registry
+- update of build image is manual
+
+## Proposed Improvements
```

Comment:
what alternatives have been considered? Have you looked into travis/circleci/github actions? Are there any other automation services that are worth considering?

I'd like to see how they compare to each other in terms of what we are trying to achieve. I.e. advantages / disadvantages.

Reply:
At the moment I'm trying to evolve what we have in CircleCI, and enable GitHub Actions in parallel.

Reply:
Also, I have used TravisCI many times in the past, it is largely the same as CricleCI, except that the UI is a little cleaner. There is no significant proposition for a swap. GitHub Action is being proposed as it offered many additional features that go beyond just running builds.

Reply:
Ok, perhaps, but we haven't made that decision here. We are only adding GHA
as an experiment, in parallel to CircleCI. We can get back to this document
once we have more info to make such decision.

Reply:
I think it is worth comparing them so we can (when the time comes) justify the decision we are making of moving to GHA, if it turns out to be the consensus then.

Reply:
I'd rather avoid this for now, the document was meant to be lightweight.